### PR TITLE
Fix for unreadable stream

### LINKED ISSUE #82: TextSerializationWriter results in System.Net.Http.HttpRequestException
```
An exception of type 'System.Net.Http.HttpRequestException' occurred in System.Private.CoreLib.dll but was not handled in user code: 'Error while copying content to a stream.'
 Inner exceptions found, see $exception in variables window for more details.
 Innermost exception 	 System.ObjectDisposedException : Cannot access a closed Stream.
   at System.ThrowHelper.ThrowObjectDisposedException_StreamClosed(String objectName)
   at System.IO.MemoryStream.CopyToAsync(Stream destination, Int32 bufferSize, CancellationToken cancellationToken)
   at System.Net.Http.StreamToStreamCopy.CopyAsync(Stream source, Stream destination, Int32 bufferSize, Boolean disposeSource, CancellationToken cancellationToken)
--- End of stack trace from previous location ---
   at System.Net.Http.HttpContent.<<CopyToAsync>g__WaitAsync|56_0>d.MoveNext()
```

Cause:
https://github.com/microsoft/kiota-serialization-text-dotnet/blob/e37cdf88af1f9e6e836c44a52faa1e5c02c91684/src/TextSerializationWriter.cs#L33-L38

- `_stream?.Dispose()` closes the stream.
- `writer?.Dispose()` also closes the stream (so just removing the `_stream?.Dispose()` call won't suffice)

Fix:

- Remove `_stream?.Dispose()` line.
- In `TextSerializationWriter` constructor, pass `leaveOpen: true` to the constructor of the `StreamWriter`.:

    ```csharp
    writer = new(_stream, leaveOpen: true);
    ```

### PR DESCRIPTION
**Pull request by jasonjoh**
Fixes #82

### CODE REVIEW

File: src/TextSerializationWriter.cs [hunk 1]
```diff
@@ -20,7 +20,13 @@ public class TextSerializationWriter : ISerializationWriter, IDisposable {
     /// </summary>
     public TextSerializationWriter()
     {
-        writer = new(_stream);
+        writer = new(
+            _stream,
+            // Default encoding
+            encoding: System.Text.Encoding.UTF8,
+            // Default buffer size
+            bufferSize: 1024,
```
Comment (baywet) [inline src/TextSerializationWriter.cs:28]:
```suggestion
```

using the named parameter and the default values, can't you just pass leave open?

**Review by baywet — DISMISSED** [review]
Thanks for the contribution.
+ changelog entry ( patch version, tomorrow)
+ csproj version bump (patch)

**Comment by andrueastman** [comment]
> @andrueastman I'm not sure why ADO is complaining here, would you mind having a look when you have a couple of minutes please?

Merging https://github.com/microsoft/kiota-serialization-text-dotnet/pull/84 into this PR/branch should fix the release task on ADO.

**Comment by baywet** [comment]
@andrueastman thanks for looking into the unit tests failing. Now it seems the path to the coverage results is wrong or the coverage results are not being generate for sonarcloud to ingest. Can you look into that as well please?

File: src/TextSerializationWriter.cs (hunk 1 above)
Comment (jasonjoh) [reply src/TextSerializationWriter.cs:28]: Yes in .NET, but .NET Framework build (part of your CI pipeline) failed. I guess .NET FX doesn't support named parameters to skip optional params.

Comment (andrueastman) [reply src/TextSerializationWriter.cs:28]: Resolving this for now as the parameters are not optional in netstandard2.0. 
https://learn.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=netstandard-2.0#system-io-streamwriter-ctor(system-io-stream-system-text-encoding-system-int32-system-boolean)

**Comment by andrueastman** [comment]
> @andrueastman thanks for looking into the unit tests failing. Now it seems the path to the coverage results is wrong or the coverage results are not being generate for sonarcloud to ingest. Can you look into that as well please?

Thanks for highlighting this. Resolved via https://github.com/microsoft/kiota-serialization-text-dotnet/pull/83/commits/2d7573d6f593efad7a657fc2411e0d1de3dee576

**Review by andrueastman — APPROVED** [review]

**Pull request merged**